File: 322-coin-change/coin-change.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        mapper = [amount + 1] * (amount+1)
        mapper[0]=0
        for i in range(amount+1):
            for c in coins:
                if i-c >= 0:
                    mapper[i] = min(mapper[i], 1 + mapper[i-c])
        return mapper[amount] if mapper[amount] != amount + 1 else -1

File: 322-coin-change/coin-change.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        int* mapper = new int[amount+1]();
        fill_n(mapper, amount+1, amount+1);
        mapper[0] = 0;
        for (int i = 0; i < amount+1; i++)
            for (int coin: coins){
                if (i - coin >= 0){
                    mapper[i] = min(mapper[i], 1 + mapper[i-coin]);
                }
            }
        return mapper[amount] == amount + 1 ? -1 : mapper[amount];
    }
};